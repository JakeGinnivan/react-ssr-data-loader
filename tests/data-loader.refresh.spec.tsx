import ComponentFixture from './helpers/component-fixture'

describe('data-loader', () => {
    it('can refresh data', async () => {
        const sut = new ComponentFixture(undefined, "testKey", { isServerSideRender: false })

        await sut.testDataPromise.resolve({ result: 'Test' })
        sut.assertState()
        sut.refreshData()

        sut.assertState()
        await sut.testDataPromise.resolve({ result: 'Test2' })
        sut.assertState()
    })

    it('calling refresh when already refreshing ignores the action', async () => {
        const sut = new ComponentFixture(undefined, "testKey", { isServerSideRender: false })

        await sut.testDataPromise.resolve({ result: 'Test' })
        sut.refreshData()
        sut.refreshData()
        sut.assertState()
    })
})