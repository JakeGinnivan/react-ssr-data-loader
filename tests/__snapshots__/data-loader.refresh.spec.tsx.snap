// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data-loader calling refresh when already refreshing ignores the action 1`] = `
Object {
  "events": Array [
    Object {
      "state": Object {
        "data": Object {},
        "loadingCount": 0,
      },
      "type": "state-changed",
    },
    Object {
      "state": Object {
        "data": Object {
          "testDataType": Object {
            "testKey": Object {
              "data": Object {
                "hasData": false,
              },
              "lastAction": Object {
                "success": true,
                "type": "none",
              },
              "status": "Fetching",
            },
          },
        },
        "loadingCount": 1,
      },
      "type": "state-changed",
    },
    Object {
      "data": Object {
        "numberLoading": 1,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "begin-loading-event",
    },
    Object {
      "state": Object {
        "data": Object {
          "testDataType": Object {
            "testKey": Object {
              "data": Object {
                "dataFromServerSideRender": false,
                "hasData": true,
                "result": Object {
                  "result": "Test",
                },
              },
              "lastAction": Object {
                "success": true,
                "type": "initial-fetch",
              },
              "status": "Idle",
            },
          },
        },
        "loadingCount": 0,
      },
      "type": "state-changed",
    },
    Object {
      "data": Object {
        "numberLoading": 1,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "end-loading-event",
    },
    Object {
      "data": Object {
        "numberLoading": 0,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "data-load-completed",
    },
    Object {
      "state": Object {
        "data": Object {
          "testDataType": Object {
            "testKey": Object {
              "data": Object {
                "dataFromServerSideRender": false,
                "hasData": true,
                "result": Object {
                  "result": "Test",
                },
              },
              "lastAction": Object {
                "success": true,
                "type": "none",
              },
              "status": "Refreshing",
            },
          },
        },
        "loadingCount": 1,
      },
      "type": "state-changed",
    },
    Object {
      "data": Object {
        "numberLoading": 1,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "begin-loading-event",
    },
  ],
  "existingData": undefined,
  "loadAllCompletedCalled": 1,
  "loadDataCount": 2,
  "renderActions": Object {
    "refresh": [Function],
    "update": [Function],
  },
  "renderCount": 3,
  "renderProps": Object {
    "data": Object {
      "dataFromServerSideRender": false,
      "hasData": true,
      "result": Object {
        "result": "Test",
      },
    },
    "lastAction": Object {
      "success": true,
      "type": "none",
    },
    "status": "Refreshing",
  },
}
`;

exports[`data-loader can refresh data 1`] = `
Object {
  "events": Array [
    Object {
      "state": Object {
        "data": Object {},
        "loadingCount": 0,
      },
      "type": "state-changed",
    },
    Object {
      "state": Object {
        "data": Object {
          "testDataType": Object {
            "testKey": Object {
              "data": Object {
                "hasData": false,
              },
              "lastAction": Object {
                "success": true,
                "type": "none",
              },
              "status": "Fetching",
            },
          },
        },
        "loadingCount": 1,
      },
      "type": "state-changed",
    },
    Object {
      "data": Object {
        "numberLoading": 1,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "begin-loading-event",
    },
    Object {
      "state": Object {
        "data": Object {
          "testDataType": Object {
            "testKey": Object {
              "data": Object {
                "dataFromServerSideRender": false,
                "hasData": true,
                "result": Object {
                  "result": "Test",
                },
              },
              "lastAction": Object {
                "success": true,
                "type": "initial-fetch",
              },
              "status": "Idle",
            },
          },
        },
        "loadingCount": 0,
      },
      "type": "state-changed",
    },
    Object {
      "data": Object {
        "numberLoading": 1,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "end-loading-event",
    },
    Object {
      "data": Object {
        "numberLoading": 0,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "data-load-completed",
    },
  ],
  "existingData": undefined,
  "loadAllCompletedCalled": 1,
  "loadDataCount": 1,
  "renderActions": Object {
    "refresh": [Function],
    "update": [Function],
  },
  "renderCount": 2,
  "renderProps": Object {
    "data": Object {
      "dataFromServerSideRender": false,
      "hasData": true,
      "result": Object {
        "result": "Test",
      },
    },
    "lastAction": Object {
      "success": true,
      "type": "initial-fetch",
    },
    "status": "Idle",
  },
}
`;

exports[`data-loader can refresh data 2`] = `
Object {
  "events": Array [
    Object {
      "state": Object {
        "data": Object {},
        "loadingCount": 0,
      },
      "type": "state-changed",
    },
    Object {
      "state": Object {
        "data": Object {
          "testDataType": Object {
            "testKey": Object {
              "data": Object {
                "hasData": false,
              },
              "lastAction": Object {
                "success": true,
                "type": "none",
              },
              "status": "Fetching",
            },
          },
        },
        "loadingCount": 1,
      },
      "type": "state-changed",
    },
    Object {
      "data": Object {
        "numberLoading": 1,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "begin-loading-event",
    },
    Object {
      "state": Object {
        "data": Object {
          "testDataType": Object {
            "testKey": Object {
              "data": Object {
                "dataFromServerSideRender": false,
                "hasData": true,
                "result": Object {
                  "result": "Test",
                },
              },
              "lastAction": Object {
                "success": true,
                "type": "initial-fetch",
              },
              "status": "Idle",
            },
          },
        },
        "loadingCount": 0,
      },
      "type": "state-changed",
    },
    Object {
      "data": Object {
        "numberLoading": 1,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "end-loading-event",
    },
    Object {
      "data": Object {
        "numberLoading": 0,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "data-load-completed",
    },
    Object {
      "state": Object {
        "data": Object {
          "testDataType": Object {
            "testKey": Object {
              "data": Object {
                "dataFromServerSideRender": false,
                "hasData": true,
                "result": Object {
                  "result": "Test",
                },
              },
              "lastAction": Object {
                "success": true,
                "type": "none",
              },
              "status": "Refreshing",
            },
          },
        },
        "loadingCount": 1,
      },
      "type": "state-changed",
    },
    Object {
      "data": Object {
        "numberLoading": 1,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "begin-loading-event",
    },
  ],
  "existingData": undefined,
  "loadAllCompletedCalled": 1,
  "loadDataCount": 2,
  "renderActions": Object {
    "refresh": [Function],
    "update": [Function],
  },
  "renderCount": 3,
  "renderProps": Object {
    "data": Object {
      "dataFromServerSideRender": false,
      "hasData": true,
      "result": Object {
        "result": "Test",
      },
    },
    "lastAction": Object {
      "success": true,
      "type": "none",
    },
    "status": "Refreshing",
  },
}
`;

exports[`data-loader can refresh data 3`] = `
Object {
  "events": Array [
    Object {
      "state": Object {
        "data": Object {},
        "loadingCount": 0,
      },
      "type": "state-changed",
    },
    Object {
      "state": Object {
        "data": Object {
          "testDataType": Object {
            "testKey": Object {
              "data": Object {
                "hasData": false,
              },
              "lastAction": Object {
                "success": true,
                "type": "none",
              },
              "status": "Fetching",
            },
          },
        },
        "loadingCount": 1,
      },
      "type": "state-changed",
    },
    Object {
      "data": Object {
        "numberLoading": 1,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "begin-loading-event",
    },
    Object {
      "state": Object {
        "data": Object {
          "testDataType": Object {
            "testKey": Object {
              "data": Object {
                "dataFromServerSideRender": false,
                "hasData": true,
                "result": Object {
                  "result": "Test",
                },
              },
              "lastAction": Object {
                "success": true,
                "type": "initial-fetch",
              },
              "status": "Idle",
            },
          },
        },
        "loadingCount": 0,
      },
      "type": "state-changed",
    },
    Object {
      "data": Object {
        "numberLoading": 1,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "end-loading-event",
    },
    Object {
      "data": Object {
        "numberLoading": 0,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "data-load-completed",
    },
    Object {
      "state": Object {
        "data": Object {
          "testDataType": Object {
            "testKey": Object {
              "data": Object {
                "dataFromServerSideRender": false,
                "hasData": true,
                "result": Object {
                  "result": "Test",
                },
              },
              "lastAction": Object {
                "success": true,
                "type": "none",
              },
              "status": "Refreshing",
            },
          },
        },
        "loadingCount": 1,
      },
      "type": "state-changed",
    },
    Object {
      "data": Object {
        "numberLoading": 1,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "begin-loading-event",
    },
    Object {
      "state": Object {
        "data": Object {
          "testDataType": Object {
            "testKey": Object {
              "data": Object {
                "dataFromServerSideRender": false,
                "hasData": true,
                "result": Object {
                  "result": "Test2",
                },
              },
              "lastAction": Object {
                "success": true,
                "type": "refresh",
              },
              "status": "Idle",
            },
          },
        },
        "loadingCount": 0,
      },
      "type": "state-changed",
    },
    Object {
      "data": Object {
        "numberLoading": 1,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "end-loading-event",
    },
    Object {
      "data": Object {
        "numberLoading": 0,
        "resourceId": "testKey",
        "resourceType": "testDataType",
      },
      "type": "data-load-completed",
    },
  ],
  "existingData": undefined,
  "loadAllCompletedCalled": 2,
  "loadDataCount": 2,
  "renderActions": Object {
    "refresh": [Function],
    "update": [Function],
  },
  "renderCount": 4,
  "renderProps": Object {
    "data": Object {
      "dataFromServerSideRender": false,
      "hasData": true,
      "result": Object {
        "result": "Test2",
      },
    },
    "lastAction": Object {
      "success": true,
      "type": "refresh",
    },
    "status": "Idle",
  },
}
`;
